src/content/datachannel/messaging/main.js

```diff
@@ -0,0 +1,169 @@
+/*
+ *  Copyright (c) 2018 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+import {LitElement, html} from 'https://unpkg.com/@polymer/lit-element?module';
+
+class MessagingSample extends LitElement {
+  constructor() {
+    super();
+    this._localConnection = this._remoteConnection = this._localChannel = this._remoteChannel = null;
+  }
+
+  disconnect(e) {
+    this._root.querySelector('#disconnectButton').disabled = true;
+    this._localConnection.close();
+    this._remoteConnection.close();
+  }
+
+  async connect(e) {
+    try {
+      this._root.querySelector('#connectButton').disabled = true;
+      const dataChannelParams = {ordered: true};
+      let lpc = this._localConnection = new RTCPeerConnection();
+      lpc.addEventListener('icecandidate', (e) => this._onIceCandidate(e, this._remoteConnection));
+      let rpc = this._remoteConnection = new RTCPeerConnection();
+      rpc.addEventListener('icecandidate', (e) => this._onIceCandidate(e, this._localConnection));
+
+      let lc = this._localChannel = this._localConnection
+        .createDataChannel('messaging-channel', dataChannelParams);
+      lc.binaryType = 'arraybuffer';
+      lc.addEventListener('open', (e) => this._localChannelOpen(e));
+      lc.addEventListener('close', (e) => this._channelClosed(e));
+      lc.addEventListener('message', (e) => this._onLocalMessageReceived(e));
+
+      rpc.addEventListener('datachannel', (e) => this._onRemoteDataChannel(e));
+
+      const localOffer = await lpc.createOffer();
+      console.log(`Got local offer ${JSON.stringify(localOffer)}`);
+      lpc.setLocalDescription(localOffer);
+      rpc.setRemoteDescription(localOffer);
+      const remoteOffer = await rpc.createAnswer();
+      console.log(`Got remote answer ${JSON.stringify(remoteOffer)}`);
+      rpc.setLocalDescription(remoteOffer);
+      lpc.setRemoteDescription(remoteOffer);
+    } catch (e) {
+      console.log(e);
+    }
+  }
+
+  async _onIceCandidate(e, connection) {
+    try {
+      console.log(`onIceCandidate: ${JSON.stringify(e)}`);
+      // eslint-disable-next-line no-unused-vars
+      const result = await connection.addIceCandidate(e.candidate);
+      console.log('addIceCandidate successful!');
+    } catch (e) {
+      console.log(`error on addIceCandidate: ${JSON.stringify(e)}`);
+    }
+  }
+
+  _localChannelOpen(event) {
+    console.log(`Local channel open: ${JSON.stringify(event)}`);
+    this._root.querySelector('#sendLocal').disabled = false;
+    this._root.querySelector('#sendRemote').disabled = false;
+    this._root.querySelector('#disconnectButton').disabled = false;
+  }
+
+  _channelClosed(event) {
+    console.log(`Channel closed: ${JSON.stringify(event)}`);
+    this._root.querySelector('#sendLocal').disabled = true;
```

Comment:
with lit-html you might want to do these the otehr way around... it have local variable that you change and just make the elements depend on that. ie don't manually modify that dom in multiple goes